nukeThread.js

```diff
@@ -116,9 +132,17 @@ function nukeThread (url) {
   return getAccessToken(query.code)
     .then(getRequester)
     .then(function (r) {
+      r.getMe().then( function(user) {
+          currentUser = user['name'];
+      });
       return getExpandedContent(r, parsedUrl);
```

Comment:
This will result in a race condition where `currentUser` will be undefined if the `getMe()` request takes awhile and finishes after the `getExpandedContent()` request, because `currentUser` will still be undefined.

Instead of using a global for `currentUser`, I think it should be passed as an argument to `deepApprove`. You can do something like this instead:

```js
return Promise.all([r.getMe(), getExpandedContent(r, parsedUrl)]).then(function ([{name}, content]) {
      if (toNuke === "nuke") {
        return deepRemove(content, document.getElementById('preserve-distinguished-checkbox').checked);
      } else {
        return deepApprove(content, document.getElementById('preserve-removed-checkbox').checked, name);
      }
})
```